sklearn/feature_extraction/text.py

```diff
@@ -401,10 +401,28 @@ def build_analyzer(self):
         preprocess = self.build_preprocessor()
 
         if self.analyzer == 'char':
+            if self.stop_words is not None:
```

Comment:
This shouldn't be duplicated, and it should also apply if analyzer is a callable

Reply:
Done .